### PR TITLE
Introducing a cluster-scoped resource in the wardle.k8s.io group.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**:
This PR adds a cluster-scoped resource to the wardle.k8s.io group.
The cluster scoped resource has a field that indicates Flunder.Names that are disallowed.
The resource is going to be used by an admission plugin.
The admission plugin will list the cluster-scope resources and check against banned names.

**Special notes for your reviewer**:
Issue: #47868

**Release note**:

```
NONE
```

### CODE REVIEW

File: test/integration/examples/apiserver_test.go [hunk 1]
```diff
@@ -448,9 +448,11 @@ func testAPIResourceList(t *testing.T, client rest.Interface) {
 		t.Fatalf("Error in unmarshalling response from server %s: %v", "/apis/wardle.k8s.io/v1alpha1", err)
 	}
 	assert.Equal(t, groupVersion.String(), apiResourceList.GroupVersion)
-	assert.Equal(t, 1, len(apiResourceList.APIResources))
-	assert.Equal(t, "flunders", apiResourceList.APIResources[0].Name)
-	assert.True(t, apiResourceList.APIResources[0].Namespaced)
+	assert.Equal(t, 2, len(apiResourceList.APIResources))
+	assert.Equal(t, "fischers", apiResourceList.APIResources[0].Name)
```
Comment (p0lyn0mial) [inline test/integration/examples/apiserver_test.go]: I think that the returned resources are ordered. Optionally we could assert against a list of expected resources.

File: staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go [hunk 2]
```diff
@@ -30,7 +30,8 @@ import (
 	"k8s.io/sample-apiserver/pkg/apis/wardle"
 	"k8s.io/sample-apiserver/pkg/apis/wardle/install"
 	"k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1"
-	wardlestorage "k8s.io/sample-apiserver/pkg/registry/wardle"
+	fischerstorage "k8s.io/sample-apiserver/pkg/registry/wardle/fischer"
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go]: I broke up wardle pkg into two(per resource).

File: staging/src/k8s.io/sample-apiserver/pkg/apis/wardle/install/install.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ func Install(groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *r
 	if err := announced.NewGroupMetaFactory(
 		&announced.GroupMetaFactoryArgs{
 			GroupName:                  wardle.GroupName,
-			RootScopedKinds:            sets.NewString("APIService"),
+			RootScopedKinds:            sets.NewString("Fischer", "FischerList"),
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/sample-apiserver/pkg/apis/wardle/install/install.go]: It seems like `APIService` was unnecessary.

**Comment by p0lyn0mial** [comment]
@deads2k FYI.

File: staging/src/k8s.io/sample-apiserver/pkg/apis/wardle/install/install.go (hunk 3 above)
Comment (deads2k) [inline staging/src/k8s.io/sample-apiserver/pkg/apis/wardle/install/install.go]: > It seems like APIService was unnecessary.

Yeah, bad copy/paste

File: staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go (hunk 2 above)
Comment (deads2k) [inline staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go]: > I broke up wardle pkg into two(per resource).

good call.

File: staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go [hunk 4]
```diff
@@ -99,10 +100,20 @@ func (c completedConfig) New() (*WardleServer, error) {
 		GenericAPIServer: genericServer,
 	}
 
+	flunderREST, err := flunderstorage.NewREST(Scheme, c.GenericConfig.RESTOptionsGetter)
```
Comment (deads2k) [inline staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go]: @sttts @p0lyn0mial Note to us.  We may want a `MustREST` (or `RESTInPeace`?) function that panics on an error just to make this look prettier.

File: staging/src/k8s.io/sample-apiserver/pkg/registry/registry.go [hunk 5]
```diff
@@ -68,28 +65,4 @@ func (apiServerStrategy) Canonicalize(obj runtime.Object) {
 
 func (apiServerStrategy) ValidateUpdate(ctx genericapirequest.Context, obj, old runtime.Object) field.ErrorList {
```
Comment (deads2k) [inline staging/src/k8s.io/sample-apiserver/pkg/registry/registry.go]: This doesn't look quite right.  I think the strategies should be per-resource.

**Comment by deads2k** [comment]
We should keep an eye on https://github.com/kubernetes/sample-apiserver/issues/7.  When @maleck13 makes integration tests work, we should add testing there.

@p0lyn0mial either you're msising some client-gen or we need to update our update scripts to generate one.  If the scripts need updating, we can open and issue and deal with it as a followup.

**Comment by deads2k** [comment]
@k8s-bot ok to test

**Comment by p0lyn0mial** [comment]
>@p0lyn0mial either you're msising some client-gen or we need to update our update scripts to generate one. If the scripts need updating, we can open and issue and deal with it as a followup.

I was playing with client-set generation. I decided that I will add them in a follow up.
Thus my working tree is polluted and by accident I added irrelevant file "//pkg/client/clientset_generated/internalclientset/typed/wardle/v1alpha1:all-"

but bazel was not deceived.

**Comment by deads2k** [comment]
> I was playing with client-set generation. I decided that I will add them in a follow up.

Makes sense. Open an issue so we don't lose it?



> but bazel was not deceived.

:)

**Comment by p0lyn0mial** [comment]
> Makes sense. Open an issue so we don't lose it?

We don't lose it, client-set will be required by an admission plugin that I am about to implement.

File: staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go (hunk 4 above)
Comment (p0lyn0mial) [inline staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go]: I am okay with `RESTInPeace` function that will panic inside.

**Comment by p0lyn0mial** [comment]
I am running `hack/update-bazel.sh` but this does not make bazel happy.

File: staging/src/k8s.io/sample-apiserver/pkg/registry/registry.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+
+// rest implements a RESTStorage for API services against etcd
+type REST struct {
```
Comment (deads2k) [inline staging/src/k8s.io/sample-apiserver/pkg/registry/registry.go]: This is the spot I expected `RESTInPeace(rest.StandardStorage, error) rest.StandardStorage` to be here.  Then each individual registry still has a `NewREST(scheme *runtime.Scheme, optsGetter generic.RESTOptionsGetter) *registry.REST)` which can be wrapped.

**Comment by deads2k** [comment]
One comment to tweak the `NewREST` creation methods to support both an error and non-error path, but this is a big enough improvement as-is that I'll trust you for a follow-up.

/lgtm

**Pull request merged**